stats/read.go

```diff
@@ -0,0 +1,215 @@
+package stats
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"gopkg.in/yaml.v2"
+
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv5"
+	"github.com/status-im/status-go/geth/common"
+)
+
+var (
+	messageHeader   = regexp.MustCompile(`INFO \[\d+-\d+\|\d+:\d+:\d+\]\sMessage delivery notification`)
+	incomingMessage = "IncomingMessage"
+	outgoingMessage = "OutgoingMessage"
+)
+
+// Timeline contains timeline and status string for an aggregated message.
+type Timeline struct {
+	When   time.Time `json:"when" toml:"when" yaml:"when" csv:"when"`
+	Status string    `json:"status" toml:"status" yaml:"status" csv:"status"`
+	Error  string    `json:"reason,omitempty" toml:"reason,omitempty" yaml:"error,omitempty" csv:"error,omitempty"`
+}
+
+// AggregatedMessage contains aggregated facts for a given message envelope and it's total
+// transition.
+type AggregatedMessage struct {
+	Envelope   string             `json:"envelope" toml:"envelope" yaml:"envelope" csv:"envelope"`
+	Protocol   string             `json:"protocol" toml:"protocol" yaml:"protocol" csv:"protocol"`
+	FromDevice string             `json:"from_device" toml:"from_device" yaml:"from_device" csv:"from_device"`
+	ToDevice   string             `json:"to_device" toml:"to_device" yaml:"to_device" csv:"to_device"`
+	Payload    string             `json:"payload" toml:"payload" yaml:"payload" csv:"payload"`
+	Direction  string             `json:"direction" toml:"direction" yaml:"direction" csv:"direction"`
+	Sent       time.Time          `json:"sent" toml:"sent" yaml:"sent" csv:"sent"`
```

Comment:
`send_time` maybe? `Sent` is too general and as an adjective, it's connected with boolean data type.